File: csharp-solution/E-CommerceSystem/Models/ShippingService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace E_CommerceSystem.Models
{
  public class ShippingService
  {
    public static void Ship(List<IShippable>items)
    {
      Dictionary<string,int>count=new Dictionary<string,int>();
      double totalWeight = 0;

      foreach (var item in items)
      {
        if(count.ContainsKey(item.GetName()))
          count[item.GetName()]++;
        else 
          count.Add(item.GetName(), 1);
        totalWeight += item.GetWeight();
      }

      Console.WriteLine("** Shipment notice **");
      Console.WriteLine($"{"Qty",- 4} {"Product",-20} {"Weight"}");

      foreach (var item in count)
      {
        double weight = items.FirstOrDefault(x => x.GetName() == item.Key).GetWeight();
        Console.WriteLine($"{item.Value,-4} {item.Key,-20} {weight:f2}g");
      }

      Console.WriteLine($"Total package weight {totalWeight / 1000.0}Kg");

    }
  }
}

File: java-solution/src/service/ShippingService.java
package service;

import interfaces.IShippable;

import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

public class ShippingService {
    public static void ship(List<IShippable>items){
        double totalWeight=0.0;
        Map<String,Integer> count=new LinkedHashMap<>();
        Map<String,Double> weight=new LinkedHashMap<>();

        for(IShippable item :items){
            count.put(item.getName(),count.getOrDefault(item.getName(),0)+1);
            weight.put(item.getName(),item.getWeight());
            totalWeight+=item.getWeight();
        }

        System.out.println("** Shipment notice **");

        for (String name :count.keySet()){
            System.out.printf("%-4d %-20s %6.0fg%n",count.get(name),name,weight.get(name));
        }

        System.out.printf("Total package weight %.1fkg%n", totalWeight / 1000);

    }
}
